doc/arch/adr-010-learn-to-rank.md

```diff
@@ -0,0 +1,297 @@
+benefit from a stronger title boost than other document types, for
+instance. Expressing that just with Elasticsearch would be prohibitively
+complex, and be almost impossible to maintain.
+
+### Validating Learning To Rank
+
+We assumed that we would be able to get better results using Learning
+To Rank. To validate this assumption, we worked on a spike.
+
+In a [validation spike][] we found that we could build a model that provided
+a better nDCG score than our current search algorithm for the queries we
+have manually gathered relevancy judgements for.
+
+This validated for us that Learning To Rank is feasible and could
+lead to more relevant results for users.
+
+Though Learning To Rank won't make relevance tuning easier, it could
+make search results more relevant.
+
+### Final decision
+
+We feel confident about implementing Learning To Rank (LTR) for the
+following reasons:
+
+- We think that it will improve search relevancy for citizens, enabling
+  them to find what they need on GOV.UK
+- We have validated our assumptions with a spike
+- LTR is a well trodden path, and other organisations have written about
+  their implementations with varying levels of openness (e.g. Wikimedia)
+- GOV.UK has implemented other machine learning projects with success
+  (related links, topic taxonomy), and we have a data science team which
+  the search team has already worked with
+- TensorFlow provides an off-the-shelf ranking module, which solves
+  some of the machine learning problems of LTR for us
+- Manual relevancy tuning is not scalable, and won't get us to the same
+  level of relevancy as LTR
+
+## Implementation plan
+
+We have decided to implement Learning To Rank by following on from
+the work started in the [validation spike][].
+
+As we are at the beginning of the project, this section may not
+accurately reflect the final implementation.
+
+[validation spike]: https://github.com/alphagov/search-api/pull/1768
+
+### Machine learning library
+
+We have decided to use [TensorFlow's ranking library][] for our
+implementation of Learning To Rank.
+
+TensorFlow is well supported, and the ranking library is under active development.
+
+We considered using the [elasticsearch-learning-to-rank][] plugin.
+However, AWS Elasticsearch does not permit us to install custom plugins.
+
+We also considered PyTorch. However, this would require writing much more
+custom machine learning code, as PyTorch don't currently have a ranking library.
+
+[TensorFlow's ranking library]: https://github.com/tensorflow/ranking
+[elasticsearch-learning-to-rank]: https://github.com/o19s/elasticsearch-learning-to-rank
+
+### Language
+
+We would like to keep the code for this project in Ruby as much as
+possible. Search API is written in Ruby and it is the primary language
+used at GOV.UK. Necessarily some machine learning code will need to be
+written in Python because TensorFlow is a Python library.
+
+### Dockerised model
```

Comment:
@kevindew sorry, that part is lacking some detail. Docker is suggested here because we're using [TensorFlow Serving](https://github.com/tensorflow/serving) which recommends using Docker.